[DOCS] Adds the 8.3.0 release notes (#134855)
elastic/kibana: docs/CHANGELOG.asciidoc

[float]
[[breaking-changes-8.3.0]]
=== Breaking change

Breaking changes can prevent your application from optimal operation and performance.
Before you upgrade, review the breaking change, then mitigate the impact to your application.

// tag::notable-breaking-changes[]

[discrete]
[[breaking-129581]]
.Removes Quandl and Graphite integrations
[%collapsible]
====
*Details* +
The experimental `.quandl` and `.graphite` functions and advanced settings are removed from *Timelion*. For more information, check {kibana-pull}129581[#129581].

*Impact* +
When you use the `vis_type_timelion.graphiteUrls` kibana.yml setting, {kib} successfully starts, but logs a `[WARN ][config.deprecation] You no longer need to configure "vis_type_timelion.graphiteUrls".` warning.

To leave your feedback about the removal of `.quandl` and `.graphite`, go to the link:https://discuss.elastic.co/c/elastic-stack/kibana/7[discuss forum].
====
      

> // end::notable-breaking-changes[]

Reviewer: ```suggestion
[discrete]
[[breaking-134855]]
.Makes Osquery All with All base privilege
[%collapsible]
====
*Details* +
The Osquery {kib} privilege has been updated, so that when the *Privileges for all features level* is set to *All*, this now applies *All* to Osquery privileges as well. Previously, users had to choose the *Customize* option to grant any access to Osquery. For more information, refer to {kibana-pull}130523[#130523].

*Impact* +
This impacts user roles that have *Privileges for all features* set to *All*. After this update, users with this role will have access to the Osquery page in {kib}. However, to use the Osquery feature fully, these requirements remain the same: users also need Read access to the logs-osquery_manager.result* index and the Osquery Manager integration must be deployed to Elastic Agents.
====

// end::notable-breaking-changes[]
```

Discover::
* Adds the ability to edit dataView, query, & filters {kibana-pull}131688[#131688]
* Extend Elasticsearch query rule with search source based data fetching {kibana-pull}124534[#124534]

Elastic Security::
For the Elastic Security 8.3.0 release information, refer to {security-guide}/release-notes.html[_Elastic Security Solution Release Notes_].

Fleet::
Changes to agent upgrade modal to allow for rolling upgrades {kibana-pull}132421[#132421]

Lens & Visualizations::
* Adds method to re-link visualizations with missing `SavedSearch` {kibana-pull}132729[#132729]
* Adds support of Data View switching for Agg-Based visualizations {kibana-pull}132184[#132184]
* Adds `showTooltip` argument {kibana-pull}130530[#130530]
* Adds `orderBucketsBySum` {kibana-pull}130457[#130457]
* Adds lines visibility for lines chart {kibana-pull}130452[#130452]
* Adds `pointsRadius`, `showPoints`, and `lineWidth` {kibana-pull}130391[#130391]
* Adds mark size configuration {kibana-pull}130361[#130361]
* Adds `xyVis` split chart {kibana-pull}130229[#130229]

Machine Learning::
* Adds the ability to create anomaly detection jobs from Lens visualizations {kibana-pull}129762[#129762]

> * Additional testing of trained model in UI {kibana-pull}129209[#129209]

Reviewer 2: ```suggestion
* Adds trained model testing for additional pytorch models {kibana-pull}129209[#129209]
```

Management::
* Adds saved object relationships to data view management {kibana-pull}132385[#132385]
* Adds support for feature_states {kibana-pull}131310[#131310]

Monitoring::
Adds the Stack monitoring health API {kibana-pull}132705[#132705]

Observability::
* Adds the ability to bulk attach multiple alerts to a Case {kibana-pull}130958[#130958]
* Adds rule details page {kibana-pull}130330[#130330]
* Adds span link {kibana-pull}126630[#126630]
* Adds ML expected model bounds as an option to Comparison controls {kibana-pull}132456[#132456]

Platform::
Adds `xyVis` and `layeredXyVis` {kibana-pull}128255[#128255]

Querying & Filtering::
Improves the current filter/search experience {kibana-pull}128401[#128401]

Security::
* Adds a manual refresh button {kibana-pull}132766[#132766]
* Adds the metadata tab {kibana-pull}131465[#131465]

> * Adds the ability to use {kib} feature privileges with API keys {kibana-pull}130918[#130918]

Reviewer 3: (we decided to not include in this in the release notes, it's too internal - I removed the release note label from PR).
```suggestion
```

====
      
[discrete]
[[deprecation-122075]]
.Deprecates `xpack.data_enhanced.*`
[%collapsible]
====
*Details* +
In kibana.yml, the `xpack.data_enhanced.*` setting is deprecated. For more information, check {kibana-pull}122075[#122075].

*Impact* +
Use the `data.*` configuration parameters instead.
====
      
[float]
[[features-8.3.0]]
=== Features

{kib} 8.3.0 adds the following new and notable features.

Alerting::
* Adds circuit breaker for max number of actions by connector type {kibana-pull}128319[#128319]
* Adds `bulkEdit` method to alerting rulesClient and internal _bulk_edit API, that allow bulk editing of rules {kibana-pull}126904[#126904]

Cases::
* Adds average time to close metric in Cases {kibana-pull}131909[#131909]
* View all alerts attached to a case in the alerts table. The feature is experimental {kibana-pull}131883[#131883]
* Adds severity field to Cases {kibana-pull}131626[#131626]
* Adds the ability to delete comments in Cases {kibana-pull}130254[#130254]

Dashboard::
Enables the new controls by default {kibana-pull}131341[#131341]

Discover::
* Adds the ability to edit dataView, query, & filters {kibana-pull}131688[#131688]

> * Extend Elasticsearch query rule with search source based data fetching {kibana-pull}124534[#124534]

Reviewer 4: This PRs were part of the initiative to enable threshold alerts in Discover, details here:

https://github.com/elastic/kibana/issues/117532

Should we create a high level issue to sum up the change? this is a significant change in Discover we should highlight

Author: @kertal are both #131688 and #124534 part of the efforts to enable threshold Alerts in Discover?

Reviewer 4: Yes, exactly!

Reviewer 4: works for me, thx!

Elastic Security::
For the Elastic Security 8.3.0 release information, refer to {security-guide}/release-notes.html[_Elastic Security Solution Release Notes_].

Fleet::
Changes to agent upgrade modal to allow for rolling upgrades {kibana-pull}132421[#132421]

Lens & Visualizations::
* Adds method to re-link visualizations with missing `SavedSearch` {kibana-pull}132729[#132729]
* Adds support of Data View switching for Agg-Based visualizations {kibana-pull}132184[#132184]
* Adds `showTooltip` argument {kibana-pull}130530[#130530]
* Adds `orderBucketsBySum` {kibana-pull}130457[#130457]
* Adds lines visibility for lines chart {kibana-pull}130452[#130452]
* Adds `pointsRadius`, `showPoints`, and `lineWidth` {kibana-pull}130391[#130391]
* Adds mark size configuration {kibana-pull}130361[#130361]
* Adds `xyVis` split chart {kibana-pull}130229[#130229]

Machine Learning::

> * Adds ML expected model bounds as an option to Comparison controls {kibana-pull}132456[#132456]

Reviewer 5: This one should be moved into the Observability section as it is in the APM section to Kibana.

* Sync panels tooltips on dashboard level {kibana-pull}130449[#130449]

Querying & Filtering::
* Support fields custom label on filter editor {kibana-pull}130533[#130533]
* Allows comma delimiter on the filters multiple selections dropdowns {kibana-pull}130266[#130266]

Security::
* Warn & disallow creating role with existing name {kibana-pull}132218[#132218]
* Improves Sessions tab {kibana-pull}131583[#131583]
* Adds experimental `csp.disableUnsafeEval` config option to remove the unsafe-eval CSP {kibana-pull}124484[#124484]

[float]
[[fixes-v8.3.0]]
=== Bug fixes
Alerting::
* Don't load connectors and connector types when there isn't an encryptionKey {kibana-pull}133335[#133335]
* Adds cloud icon "ess-icon" at the end of the config keys in "alerting" {kibana-pull}131735[#131735]
* Fixes optional spaceId in rules_client {kibana-pull}130704[#130704]

Dashboard::
* Adds Fatal Error Handling {kibana-pull}133579[#133579]
* Hide in Print Mode {kibana-pull}133446[#133446]
* Send Control State to Reporting Via Locator {kibana-pull}133425[#133425]
* Fixes new controls causing unsaved changes bug {kibana-pull}132850[#132850]

> * Fixes the listingLimit settings url {kibana-pull}129701[#129701]

Reviewer 6: Wondering if this should go under the `dashboard` section? It's a fix for the generic listing page, so it applies to Maps, Visualize, Graph, and Dashboard. Maybe it should go under the management section

* Make suggestions depend on active data in *Lens* {kibana-pull}129326[#129326]
* Adds back setMapView function in *Vega* {kibana-pull}128914[#128914]

Machine Learning::
* Fixes creation of the custom URLs for Kibana Dashboard {kibana-pull}134248[#134248]
* Fixes expanded row stats not loading all correctly whenever sort by cardinality {kibana-pull}134113[#134113]
* Fixes Data visualizer showing 0 count in the doc count chart even though documents do exist {kibana-pull}134083[#134083]
* Fixes querying anomalies for the Single Metric Viewer {kibana-pull}133419[#133419]
* Fixes Anomaly Charts filtering based on the swim lane job selection {kibana-pull}133299[#133299]
* Fixes handling of unrecognised URLs {kibana-pull}133157[#133157]
* Prevent duplicate field selection in detector modal {kibana-pull}133018[#133018]
* Fixes single metric job with doc_count field {kibana-pull}132997[#132997]
* Hide job messages clear notifications tooltip on click {kibana-pull}132982[#132982]
* Filtering category runtime fields in advanced wizard {kibana-pull}132916[#132916]
* Fixes trained model testing so it is available for users with ML read permissions {kibana-pull}132698[#132698]

> * Wizard validation improvements {kibana-pull}132615[#132615]

Reviewer 5: This is already in the enhancements section so should be removed from here.

* Hide create filter UI in canvas {kibana-pull}133943[#133943]
* Use label features from ES vector tile search API to fix multiple labels {kibana-pull}132080[#132080]
* Fixes Map panels should not show the user controls in a dashboard report {kibana-pull}131970[#131970]
* Show "no results" found for vector tile aggregations when there are no results {kibana-pull}130821[#130821]

Monitoring::
* Prevent exceptions in rule when no data present {kibana-pull}131332[#131332]
* Fixes displaying ES version for external collection {kibana-pull}131194[#131194]
* Fixes node type detection for external collection {kibana-pull}131156[#131156]
* Use server.publicBaseUrl in Alert links {kibana-pull}131154[#131154]

Observability::
* Fixes x-axis on error charts {kibana-pull}134193[#134193]
* Display ENVIRONMENT_ALL label instead of value {kibana-pull}133616[#133616]
* Fixes normalizers to not parse list values if they are already parsed {kibana-pull}133563[#133563]
* Change bucket_scripts to use params for thresholds {kibana-pull}133214[#133214]
* Use Observability rule type registry for list of rule types {kibana-pull}132484[#132484]

> * APM anomaly rule type should appear in o11y rules page {kibana-pull}132476[#132476]

Reviewer 7: ```suggestion
* APM anomaly rule type should appear in observability rules page {kibana-pull}132476[#132476]
```

* Fixes monitors details page errors {kibana-pull}132196[#132196]

> * Set a valid service_name for python APM onboarding {kibana-pull}131959[#131959]

Reviewer 7: ```suggestion
* Set a valid `service_name` for python APM onboarding {kibana-pull}131959[#131959]
```

* Rename service groups template titles and links {kibana-pull}131381[#131381]
* Show service group icon only for service groups {kibana-pull}131138[#131138]
* Refactor Metric Threshold rule to push evaluations to Elasticsearch {kibana-pull}126214[#126214]
* Ellipsis truncation issue - dependencies and service section {kibana-pull}122203[#122203]
* Fixes lookback window for anomalies for anomaly alert {kibana-pull}93389[#93389]

Operations::
Fixes error handling on precommit hook {kibana-pull}132998[#132998]

Platform::

> Retry INIT step when hitting incompatible_cluster_routing_allocation {kibana-pull}131809[#131809]

Reviewer 8: It might be more user-friendly to rephrase this in a less technical way:
```suggestion
Prevents Kibana from bootlooping during migrations when Elasticsearch routing allocation settings are incompatible {kibana-pull}131809[#131809]
```

Cases::
* Adds average time to close metric in Cases {kibana-pull}131909[#131909]
* View all alerts attached to a case in the alerts table. The feature is experimental {kibana-pull}131883[#131883]
* Adds severity field to Cases {kibana-pull}131626[#131626]
* Adds the ability to delete comments in Cases {kibana-pull}130254[#130254]

Dashboard::
Enables the new controls by default {kibana-pull}131341[#131341]

Discover::
* To enable Threshold Alerts, adds the ability to edit dataView, query, & filters {kibana-pull}131688[#131688]
* To enable Threshold Alerts, extended the {es} query rule with search source-based data fetching {kibana-pull}124534[#124534]

Elastic Security::
For the Elastic Security 8.3.0 release information, refer to {security-guide}/release-notes.html[_Elastic Security Solution Release Notes_].

Fleet::
Changes to agent upgrade modal to allow for rolling upgrades {kibana-pull}132421[#132421]

Lens & Visualizations::
* Adds method to re-link visualizations with missing `SavedSearch` {kibana-pull}132729[#132729]
* Adds support of Data View switching for Agg-Based visualizations {kibana-pull}132184[#132184]

> * Adds `showTooltip` argument {kibana-pull}130530[#130530]
> * Adds `orderBucketsBySum` {kibana-pull}130457[#130457]
> * Adds lines visibility for lines chart {kibana-pull}130452[#130452]
> * Adds `pointsRadius`, `showPoints`, and `lineWidth` {kibana-pull}130391[#130391]
> * Adds mark size configuration {kibana-pull}130361[#130361]
> * Adds `xyVis` split chart {kibana-pull}130229[#130229]

Reviewer: ```suggestion
```

* Adds the ability to create a case from within the selection case modal {kibana-pull}128882[#128882]

Content Management::

> The list view for Dashboard, Visualize Library, Maps, and Graph has a new "Last updated" column to easily access content that has been recently modified {kibana-pull}132321[#132321]

Reviewer 9: 🎉
